src/all/consumybili/src/eu/kanade/tachiyomi/animeextension/all/consumybili/ConsumyBili.kt

```diff
@@ -0,0 +1,874 @@
+
+        val animeList = parsed.data.Page.media.map { ani ->
+            SAnime.create().apply {
+                title = ani.title.romaji
+                thumbnail_url = ani.coverImage.extraLarge
+                setUrlWithoutDomain(
+                    AnimeInfo(
+                        title = ani.title.romaji,
+                        id = ani.id.toString(),
+                        thumbnailUrl = ani.coverImage.extraLarge,
+                        studio = ani.studios.nodes.firstOrNull()?.name ?: "",
+                        genres = ani.genres.joinToString(", "),
+                        description = ani.description,
+                        status = ani.status
+                    ).toJsonString()
+                )
+            }
+        }
+        return AnimesPage(animeList, parsed.data.Page.pageInfo.hasNextPage)
+    }
+
+    // =============================== Latest ===============================
+
+    override fun latestUpdatesRequest(page: Int): Request = throw Exception("Not used")
+
+    override fun latestUpdatesParse(response: Response): AnimesPage = throw Exception("Not used")
+
+    // =============================== Search ===============================
+
+    override fun searchAnimeRequest(page: Int, query: String, filters: AnimeFilterList): Request {
+        val bodyQuery = """
+            query (${'$'}page: Int, ${'$'}perPage: Int, ${'$'}search: String) {
+                Page(page: ${'$'}page, perPage: ${'$'}perPage) {
+                    pageInfo {
+                        hasNextPage
+                    }
+                    media(search: ${'$'}search, type: ANIME, sort: TRENDING_DESC) {
+                        id
+                        title {
+                            romaji
+                        }
+                        coverImage {
+                            extraLarge
+                        }
+                        studios (isMain: true) {
+                            nodes {
+                                name
+                            }
+                        }
+                        genres
+                        description (asHtml: true)
+                        status
+                    }
+                }
+            }
+        """.trimIndent()
+        val variables = """
+            {
+              "page": $page,
+              "perPage": 40,
+              "search": "$query"
+            }
+        """.trimIndent()
+        val requestBody = FormBody.Builder()
+            .add("query", bodyQuery)
+            .add("variables", variables)
+            .build()
+        return POST(anilistUrl, body = requestBody)
+    }
+
+    override fun searchAnimeParse(response: Response): AnimesPage = popularAnimeParse(response)
+
+    // =========================== Anime Details ============================
+
+    override fun fetchAnimeDetails(anime: SAnime): Observable<SAnime> {
+        val parsed = json.decodeFromString<AnimeInfo>(anime.url)
```

Comment:
You don't have to pass the attributes like this, you can just set them in popular & search and return empty in animeDetail